ev3/lego.py

```diff
@@ -1,23 +1,57 @@
-from .ev3dev import Msensor, Motor
+from ev3dev import LegoSensor, Motor
 
 
-class TouchSensor(Msensor):
+class TouchSensor(LegoSensor):
 
     def __init__(self, port=-1):
-        Msensor.__init__(self, port, type_id=16, name='lego-ev3-touch')
+#Both lego-nxt-touch and lego-ev3-touch support auto
+        LegoSensor.__init__(self, port, type_id=16, name='auto')
 
     @property
     def is_pushed(self):
         self.mode = 'TOUCH'
         return bool(self.value0)
 
+class LightSensor(LegoSensor):
+    
+    def __init__(self, port=-1):
+        LegoSensor.__init__(self, port, type_id=1, name='lego-nxt-light')
+    
+    @property
+    def reflect(self):
+        self.mode = 'REFLECT'
+        #Reflected light intensity (0 to 100)
+        return self.value0/(int(self.decimals)*10)
+
+    @property
+    def ambient(self):
+        self.mode = 'AMBIENT'
+        #Ambient light intensity (0 to 100)
+        return self.value0/(int(self.decimals)*10)
+
+class SoundSensor(LegoSensor):
```

Comment:
There is actually a driver for the [NXT sound sensor](http://www.ev3dev.org/docs/sensors/lego-nxt-sound-sensor/) that scales the values properly instead of using the generic analog driver.